chore: resolve JavaScript lint errors #8013

Gauravkaushik-1206 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: na
  - task: lint_package_json status: na
  - task: lint_repl_help status: na
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: na
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: na
  - task: lint_typescript_tests status: na
  - task: lint_license_headers status: passed ---

Resolves #8013 

## Description

> What is the purpose of this pull request?

This pull request:

-  Updated the onExists function to handle the callback error. 
- Replaced the new Array() with [].
## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #8013

## Questions

> Any questions for reviewers of this pull request?

The test expect us to move futher to parent diectory even if we have recieve error like path does not exist, permission denied or Invalid. So, I have just added the coment that we can log the error or handle the error and moved to next diectory(parent). does i have to log the error or handle the error ??. The test cases expect us to ignore them.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Gauravkaushik-1206 committed:
fix: resolve the javascript lint error

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/fs/resolve-parent-path-by/lib/async.js:
```diff
@@ -124,7 +124,11 @@ function resolveParentPathBy( path, options, predicate, clbk ) {
 	* @returns {void}
 	*/
-	function onExists( error, bool ) { // eslint-disable-line handle-callback-err
+	function onExists( error, bool ) {
+		if (error) {
+			// Log or handle the error
+			return next();
+		}
```
```suggestion
	function onExists( error, bool ) { // eslint-disable-line node/handle-callback-err
```

kgryte committed:
style: disable lint rule

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
docs: update example

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/ranks/examples/index.js:
@Gauravkaushik-1206 Would you mind propagating these changes to the `stats/ranks` README, which includes a code block containing a very similar example?

Gauravkaushik-1206 replied:
Sure

kgryte replied:
Thank you!

Gauravkaushik-1206 committed:
fix: update stats/ranks readme file

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (approved):
LGTM
